### ISSUE TITLE
progressbars (tqdm) in dagit

### ISSUE THREAD

**Issue by mrdrprofuroboros** [dagster-io/dagster#3071]
Some solids are quite a massive map-reduce tasks whose execution time heavily depends on the config. So I want to get an estimate of what to expect from them and the most natural way would be a progress bar. when I execute pipeline in dev mode via python API, I see tqdms running in my jupyter, but I would like to have it in dagit. the MVP here could be just tqdm plugin, yielding % messages in plain text, but of course UI is preferred

tqdm seems to be the core tool in data science, it makes computations faster :smile:

**Comment by nancydyc** [comment]
#3070

**Comment by mrdrprofuroboros** [comment]
here is a simple example
```
from time import sleep
from dagster import pipeline, solid
from tqdm.auto import tqdm

class dagster_tqdm(tqdm):
    def __init__(self, *args, **kwargs):
        kwargs = kwargs.copy()
        self.context = kwargs.pop('context')
        super().__init__(*args, **kwargs)
    def display(self, **kwargs):
        super().display(**kwargs)
        fmt = self.format_dict
        if 'bar_format' in fmt and fmt['bar_format']:
            fmt['bar_format'] = fmt['bar_format'].replace('<bar/>', '{bar}')
        else:
            fmt['bar_format'] = '{l_bar}{bar}{r_bar}'
        fmt['bar_format'] = fmt['bar_format'].replace('{bar}', '{bar:10u}')
        self.context.log.info(self.format_meter(**fmt))

@solid
def tqdm_solid(context):
    for i in dagster_tqdm(range(5), context=context, postfix='Fist action'):
        sleep(0.05)
    for i in dagster_tqdm(range(5), context=context, postfix='Second one'):
        sleep(0.05)

@pipeline
def tqdm_pipeline():
    tqdm_solid()
```
![image](https://user-images.githubusercontent.com/21140507/95633332-23601580-0a90-11eb-869b-bd2d67cc52ef.png)

**Comment by schrockn** [comment]
This could be quite a nice feature. I believe the best way to implement it would be to add a Progress event to the event metadata stream that solids could yield and that the dagit run viewer could interpret.

**Comment by ThomasRolfsnes-EP** [comment]
+1

**Comment by danielgafni** [comment]
I'm training deep learning models in Dagster (the job can take around a day to complete) and being able to see the progress from the UI would be really useful!

**Comment by schrockn** [comment]
@mjclarke94 in the interim you could just periodically log to the structured event log and it will show up in real time.

**Comment by mjclarke94** [comment]
Yeah, we embedded a `context.log.debug()` in our generator to achieve the above. This isn't something need, just a quality of life thing. I tend to reflexively wrap an iterable in tqdm at this point, so having that output be readable without drowning dagit with messages would be nice!

**Comment by danielgafni** [comment]
Seems like a long-running tqdm progress bar causes complete freezing of the Run page in Dagit #13026

**Comment by danielgafni** [comment]
By the way, a workaround for the original issue is to use tqdm_telegram or tqdm_slack

**Comment by f0rodo** [comment]
+1

**Comment by danielgafni** [comment]
It's worth noting that `tqdm` is currently very usable with Dagster: the logs are able to display live tqdm bars with in-place updates instead of spamming new lines all the time.

**Comment by medihack** [comment]
Does someone have a current example? The solution in the [above comment](https://github.com/dagster-io/dagster/issues/3071#issuecomment-706411744) does not work unfortunately ('dagster_tqdm' object as no attribute 'disable')
